File: result/translate/transcoder/java2python/python/REARRANGE_POSITIVE_AND_NEGATIVE_NUMBERS_PUBLISH.py
def f_gold ( arr , n ) :
    i , temp = - 1 , 0
    for j in range ( n ) :
        if arr [ j ] < 0 :
            i += 1
            temp = arr [ i ]
            arr [ i ] , arr [ j ] = arr [ j ] , temp
    pos , neg = i + 1 , 0
    while pos < n and neg < pos and arr [ neg ] < 0 :
        temp = arr [ neg ]
        arr [ neg ] , arr [ pos ] = arr [ pos ] , temp
        pos += 1
        neg += 2
    return arr , temp

File: result/translate/starchat/python2java/java/REARRANGE_POSITIVE_AND_NEGATIVE_NUMBERS_PUBLISH.java
public static void f_gold ( int arr [ ], int n ) {
        int i = -1;
        for ( int j = 0; j < n; j++ ) {
            if ( arr [ j ] < 0 ) {
                i++;
                int temp = arr [ i ];
                arr [ i ] = arr [ j ];
                arr [ j ] = temp;
            }
        }
        int pos = i + 1, neg = 0;
        while ( pos < n && neg < pos && arr [ neg ] < 0 ) {
            int temp = arr [ neg ];
            arr [ neg ] = arr [ pos ];
            arr [ pos ] = temp;
            pos += 1;
            neg += 2;
        }
    }